CHANGELOG.md

```diff
@@ -10,6 +10,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - [BREAKING] Removed `relativeUrl` function from `url-utils`, shifting responsibility to `Bundler#analyzer.urlResolver.relative()`.
 - [BREAKING] Removed `import-utils.ts` and migrated all HTML-specific bundling code out of it and out of `bundler.ts` into a new class `HtmlBundler`.  Renamed `ast-utils.ts` to `parse5-utils.ts` since "ast" is ambiguous when we have JavaScript ASTs coming in.
 - [BREAKING] `Bundle` class now has a required `type` property as first argument to its constructor, with current possible values of `html-fragment` and `es6-module`.
+- [BREAKING] The `DepsIndex` type returned from `buildDepsIndex` has changed to a single map of entrypoints to dependencies.
+- Added support for `<script type="module">` and bundling of ES6 modules by following `import` statements.  Dynamic `import()` statements are treated similarly to the `<link rel="lazy-import">` tag in that the import target implicitly defines a new entrypoint/bundle.
+- Shared bundles containing only one file are named after the original file instead of having naming delegated to shared bundle URL mapper.
```

Comment:
Can you explain the reasoning behind this? This sounds like something that would may people up ("why was this file skipped by the bundler?")

Reply:
Hmm.  The premise here was that it didn't make sense to essentially just rename a single file to 'shared_bundle_x.js' just because that one file was imported from two locations.  I'm realizing in thinking more about it that the single file case is actually more specific version of a broader class of 'single top-level direct dependency' expressed in the bundle, meaning that maybe 'a.js' is depended on by multiple entrypoints but 'a.js' imports 'b.js' and 'b.js' has no other direct dependents, meaning that there really could be a bundle called 'a.js' which rolls up 'b.js' into it, and that case is not handled here.

You've given me something more to think about.  Thanks!

Reply:
I removed this special-case logic and the CHANGELOG item for it.  It was one of those things that made sense in the moment but needs to be thought about and handled as a separate concern.